### PR TITLE
add FilterQuality diagram

### PR DESCRIPTION
**Pull request by yjbanov**
Add a diagram showing the effect of `FilterQuality` on scaled images.

Bonus: exit with a non-zero code when any of the worker jobs fail.

### CODE REVIEW

File: bin/generate.dart [hunk 1]
```diff
@@ -348,11 +349,33 @@ class DiagramGenerator {
...
     }
   }
 }
 
+/// Sets the exit code to 1 if at least one of the jobs failed.
+void _checkJobResults(List<WorkerJob> jobs) {
+  if (jobs.any(_hasJobFailed)) {
+    // We're already printing the error message. All that's left is set
+    // the exit code.
+    exitCode = 1;
```
Comment (Hixie) [inline bin/generate.dart:363]: can you return 1 here and have the exitCode be set outside this? In general code is easier to reason about if it does not cause changes outside of its own state.

**Review by Hixie — APPROVED** [review]
LGTM modulo minor comments
LGTM

File: bin/generate.dart (hunk 1 above)
Comment (yjbanov) [reply bin/generate.dart]: Done. Consolidated error reporting into `GeneratorException`.

Comment (Hixie) [reply bin/generate.dart]: that works. thanks.

**Pull request merged**